internal/rbd/rbd_util.go

```diff
@@ -1508,6 +1508,49 @@ func (rv *rbdVolume) RepairThickProvision() error {
 	return nil
 }
 
+// DeepCopy creates an independent image (dest) from the source image. This
+// process may take some time when the image is large.
+func (rv *rbdVolume) DeepCopy(dest *rbdVolume) error {
+	opts := librbd.NewRbdImageOptions()
+	defer opts.Destroy()
+
+	// when doing DeepCopy, also flatten the new image
```

Comment:
@nixpanic @Madhu-1 I am not sure we have to explictly do  a flattening of the new image here . iic,after deep copy the new image is flattened,in that case we dont need to do that, but I could be wrong

Reply:
That is done by passing `ImageOptionFlatten` which is done on the next line, hence this comment.